### ISSUE TITLE
Gradient Checkpointing

### ISSUE THREAD

**Issue by koleshjr** [zjysteven/lmms-finetune#71]
why are we doing this?

```
if training_args.gradient_checkpointing: 
     model.enable_input_require_grads() 
```

**Comment by zjysteven** [comment]
Honestly I don't know the very detail, but essentially it enables correct gradient computation when using gradient checkpointing.

**Comment by koleshjr** [comment]
Oh I see
